sources/Interop/Windows/winrt/wrl/client/ComPtr`1.Manual.cs

```diff
@@ -4,150 +4,401 @@
+        /// </summary>
+        /// <param name="other">The other <see cref="ComPtr{T}"/> instance to copy.</param>
         public ComPtr(ComPtr<T> other)
         {
             ptr_ = other.ptr_;
+
             InternalAddRef();
         }
 
+        /// <summary>
+        /// Converts a raw pointer to a new <see cref="ComPtr{T}"/> instance and increments the ref count.
+        /// </summary>
+        /// <param name="other">The raw pointer to wrap.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static implicit operator ComPtr<T>(T* other)
         {
             return new ComPtr<T>(other);
         }
 
+        /// <summary>
+        /// Unwraps a <see cref="ComPtr{T}"/> instance and returns the internal raw pointer.
+        /// </summary>
+        /// <param name="other">The <see cref="ComPtr{T}"/> instance to unwrap.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static implicit operator T*(ComPtr<T> other)
         {
             return other.Get();
         }
 
+        /// <summary>
+        /// Converts the current object reference to type <typeparamref name="U"/> and assigns that to a target <see cref="ComPtr{T}"/> value.
+        /// </summary>
+        /// <typeparam name="U">The interface type to use to try casting the current COM object.</typeparam>
+        /// <param name="p">A raw pointer to the target <see cref="ComPtr{T}"/> value to write to.</param>
+        /// <returns>The result of <see cref="IUnknown.QueryInterface"/> for the target type <typeparamref name="U"/>.</returns>
+        /// <remarks>This method will automatically release the target COM object pointed to by <paramref name="p"/>, if any.</remarks>
         public readonly int As<U>(ComPtr<U>* p)
             where U : unmanaged
         {
-            var iid = typeof(U).GUID;
-            return ((IUnknown*)(ptr_))->QueryInterface(&iid, (void**)(p->ReleaseAndGetAddressOf()));
+            return ((IUnknown*)ptr_)->QueryInterface(__uuidof<U>(), (void**)p->ReleaseAndGetAddressOf());
+        }
+
+        /// <summary>
+        /// Converts the current object reference to type <typeparamref name="U"/> and assigns that to a target <see cref="ComPtr{T}"/> value.
+        /// </summary>
+        /// <typeparam name="U">The interface type to use to try casting the current COM object.</typeparam>
+        /// <param name="other">A reference to the target <see cref="ComPtr{T}"/> value to write to.</param>
+        /// <returns>The result of <see cref="IUnknown.QueryInterface"/> for the target type <typeparamref name="U"/>.</returns>
+        /// <remarks>This method will automatically release the target COM object pointed to by <paramref name="other"/>, if any.</remarks>
+        public readonly int As<U>(ref ComPtr<U> other)
+            where U : unmanaged
+        {
+            fixed (ComPtr<U>* p = &other)
```

Comment:
The  `pin` could be avoided here by using a local and then internally assigning `other.ptr_`.

Reply:
Same with other cases like https://github.com/terrafx/terrafx.interop.windows/pull/151/files#diff-a1a828334f363e63b93afdd4a19f4f6ca865a85ceb9411137ecc4073be026014R122

Reply:
Same as below, removed in e10f572606582cdb1e485423992a813aecd2bba8.